internal/repositories/repositories.go

```diff
@@ -99,6 +103,38 @@ func (r *Repository) GetSignalLatest(ctx context.Context, latestArgs *model.Late
 	return coll, nil
 }
 
+// GetDeviceActivity returns device status activity level.
+func (r *Repository) GetDeviceActivity(ctx context.Context, vehicleTokenID int, adManuf string) (*model.DeviceActivity, error) {
+	mtrLower := strings.ToLower(adManuf)
+	if mtrLower == hashDogMtr {
+		mtrLower = macaronMtr
+	}
+
+	args := &model.LatestSignalsArgs{
+		IncludeLastSeen: true,
+		SignalArgs: model.SignalArgs{
+			TokenID: uint32(vehicleTokenID),
+			Filter: &model.SignalFilter{
+				Source: &mtrLower,
```

Comment:
This is source, it should be something like `dimo/integration/27qftVRWQYpVDcO5DltO5Ojbjxk`. I don't think this has any chance of working.